daemon/Makefile

```diff
@@ -4,7 +4,7 @@ include ../Makefile.defs
 TARGET=cilium-agent
 LINKS=cilium-node-monitor
 SOURCES := $(shell find ../api ../common ../daemon ../pkg . \( -name '*.go'  ! -name '*_test.go' \))
-$(TARGET): $(SOURCES)
+$(TARGET): $(SOURCES) ../Dockerfile ../Makefile ../Makefile.defs Makefile
```

Comment:
This is odd, why would a change in the Dockerfile would need to make the binary to be rebuild? Ah, is that in the case that when we are creating a Dockerimage the binary is not re-compiled in case it already exists?

Reply:
Yes. For example when updating the Envoy SHA dependency.